### PR TITLE
Add Vine HTML component

### PR DESCRIPTION
**Pull request by niallkennedy**
Add an `amp-vine` custom element. Pause video through inactive callback.

Includes example file, automated tests, and manual test file.

Intent to implement issue: https://github.com/ampproject/amphtml/issues/1005

### CODE REVIEW

File: extensions/amp-vine/0.1/amp-vine.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {loadPromise} from '../../../src/event-helper';
+
+class AmpVine extends AMP.BaseElement {
```
Comment (erwinmombay) [inline extensions/amp-vine/0.1/amp-vine.js]: add newline between class declaration an and first method.

File: examples/vine.amp.html [hunk 2]
```diff
@@ -0,0 +1,30 @@
+<!doctype html>
+<html ⚡>
+<head>
+  <meta charset="utf-8">
+  <title>Vine examples</title>
+  <link rel="canonical" href="amps.html">
+  <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
+  <link href="https://fonts.googleapis.com/css?family=Questrial" rel="stylesheet" type="text/css">
+  <script async custom-element="amp-vine" src="https://cdn.ampproject.org/v0/amp-vine-0.1.js"></script>
+  <style>body {opacity: 0}</style><noscript><style>body {opacity: 1}</style></noscript>
```
Comment (erwinmombay) [inline examples/vine.amp.html]: @dvoytenko @sriramkrish85 this will be changing soon correct?

File: extensions/amp-vine/0.1/amp-vine.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  /** @override */
+  preconnectCallback(onLayout) {
+    // the Vine iframe
+    this.preconnect.url('https://vine.co');
```
Comment (erwinmombay) [inline extensions/amp-vine/0.1/amp-vine.js]: @dvoytenko would it be appropriate to pass in the the `opt_alsoConnecting` flag here like so:

```js
this.preconnect.url('https://www.youtube.com', onLayout);
```

File: examples/vine.amp.html (hunk 2 above)
Comment (dvoytenko) [inline examples/vine.amp.html]: No. Please keep as is.

File: extensions/amp-vine/0.1/amp-vine.js (hunk 3 above)
Comment (dvoytenko) [inline extensions/amp-vine/0.1/amp-vine.js]: Yes.

Comment (erwinmombay) [inline extensions/amp-vine/0.1/amp-vine.js]: pass `onLayout` as 2nd arg to `url` method call.

**Comment by erwinmombay** [comment]
Looks great! just some minor recommendations and will LGTM.

**Comment by erwinmombay** [comment]
@niallkennedy also feel free to add yourself to CONTRIBUTORS

**Comment by niallkennedy** [comment]
@erwinmombay outstanding requests should have been met with 4072da7

**Comment by erwinmombay** [comment]
LGTM

**Comment by erwinmombay** [comment]
@cramforce should we be updating the validator in this PR too for the `amp-vine` tag?

**Comment by cramforce** [comment]
@erwinmombay I think we aren't there yet. Maybe ping @powdercloud on chat.

**Comment by erwinmombay** [comment]
@niallkennedy just one request, do you mind squashing this into 1 commit. thanks! will merge right after

**Comment by erwinmombay** [comment]
@cramforce my mistake, its already in the protoascii definition.

**Comment by niallkennedy** [comment]
@erwinmombay squashed into eb98861

**Pull request merged**